core/src/main/java/jenkins/slaves/DefaultJnlpSlaveReceiver.java

```diff
@@ -152,7 +154,7 @@ public void beforeChannel(@NonNull JnlpConnectionState event) {
         final SlaveComputer computer = state.getNode();
         final OutputStream log = computer.openLogFile();
         state.setLog(log);
-        try (PrintWriter logw = new PrintWriter(log, true)) {
+        try (PrintWriter logw = new PrintWriter(new OutputStreamWriter(log, Charset.defaultCharset()), true)) {
```

Comment:
Agent log files should be in UTF-8.

Reply:
Here and elsewhere, please provide reasoning when you make requests of me.

Reply:
Any plugin can write text to these logs, and it is intended solely for use by Jenkins, so there is no reason to allow non-UTF-8 encoding.

Looks like some other places need to be fixed; at least: https://github.com/jenkinsci/jenkins/blob/75ce81d12ca29eb4a3bc7bd3a4018d65452e90d6/core/src/main/java/hudson/slaves/SlaveComputer.java#L155 https://github.com/jenkinsci/jenkins/blob/75ce81d12ca29eb4a3bc7bd3a4018d65452e90d6/core/src/main/java/hudson/slaves/SlaveComputer.java#L623 https://github.com/jenkinsci/jenkins/blob/75ce81d12ca29eb4a3bc7bd3a4018d65452e90d6/core/src/main/java/hudson/model/Computer.java#L335 https://github.com/jenkinsci/jenkins/blob/75ce81d12ca29eb4a3bc7bd3a4018d65452e90d6/core/src/main/java/hudson/model/Computer.java#L343

Reply:
A bigger change than what I am trying to achieve in this PR. For now I have added a comment matching https://github.com/jenkinsci/jenkins/blob/75ce81d12ca29eb4a3bc7bd3a4018d65452e90d6/core/src/main/java/hudson/model/Computer.java#L335 to document the status quo.